src/diff-snapshot.js

```diff
@@ -83,6 +83,38 @@ const shouldUpdate = ({ pass, updateSnapshot, updatePassedSnapshot }) => (
   (!pass && updateSnapshot) || (pass && updatePassedSnapshot)
 );
 
+const shouldFail = ({
+  totalPixels,
+  diffPixelCount,
+  hasSizeMismatch,
+  allowSizeMismatch,
+  failureThresholdType,
+  failureThreshold,
+}) => {
+  let pass = false;
+  let diffSize = false;
+  const diffRatio = diffPixelCount / totalPixels;
+  if (hasSizeMismatch) {
+    // do not fail if allowSizeMismatch is set
+    pass = Boolean(allowSizeMismatch);
```

Comment:
Why is `Boolean` needed?

Reply:
It is not, I just like the explicitly that pass will contain a Boolean value.
```suggestion
    pass = allowSizeMismatch;
```